src/geode/inspector/topology/private/brep_blocks_topology_impl.cpp

```diff
@@ -126,15 +128,108 @@ namespace geode
                             brep_.block( block_uuid ) );
                     } );
+
+                const auto nb_boundary_surface_cmvs = count_cmvs( surface_cmvs,
+                    [&block_uuid, this]( const ComponentMeshVertex& cmv ) {
+                        return this->brep_.is_boundary(
+                            brep_.surface( cmv.component_id.id() ),
+                            brep_.block( block_uuid ) );
+                    } );
+                const auto nb_boundary_line_cmvs = count_cmvs( line_cmvs,
+                    [&block_uuid, this]( const ComponentMeshVertex& cmv ) {
+                        for( const auto& block_boundary :
+                            this->brep_.boundaries(
+                                brep_.block( block_uuid ) ) )
+                        {
+                            for( const auto& surface_boundary :
+                                this->brep_.boundaries( block_boundary ) )
+                            {
+                                if( surface_boundary.id()
+                                    == cmv.component_id.id() )
+                                {
+                                    return true;
+                                }
+                            }
+                            for( const auto& surface_internal :
+                                this->brep_.internal_lines( block_boundary ) )
+                            {
+                                if( surface_internal.id()
+                                    == cmv.component_id.id() )
+                                {
+                                    return true;
+                                }
+                            }
+                        }
+                        return false;
+                    } );
+                if( corner_cmvs.size() == 1 )
+                {
+                    if( nb_boundary_line_cmvs == 1 )
+                    {
+                        if( nb_block_cmvs != 1 )
+                        {
+                            if( verbose_ )
+                            {
+                                Logger::info( "Unique vertex with index ",
+                                    unique_vertex_index,
+                                    " is part of the block ",
+                                    block_uuid.string(),
+                                    " and exactly one corner and one line but "
+                                    "has ",
+                                    nb_block_cmvs,
+                                    " block component mesh vertices (should be "
+                                    "1)." );
+                            }
+                            return false;
+                        }
+                        continue;
+                    }
+
+                    if( nb_internal_surface_cmvs != 0 )
+                    {
+                        continue;
+                    }
```

Comment:
I think in this case we want to skip the rest of the if(cornet_cmvs) but not the following tests -> maybe move it in the first if, wihch would become if( corner_cmvs.size() == 1 && nb_internal_surface_cmvs==0)

Reply:
You are right